### PR TITLE
Fix fotmob get match players

### LINKED ISSUE #277: Error in fotmob_get_match_players()
Getting an error with the below code. Was working as recently as yesterday.
`
fotmob_get_match_players(2999743)

Error: ℹ In index: 1.
Caused by error in `map2()`:
ℹ In index: 3.
Caused by error in `map()`:
ℹ In index: 1.
Caused by error in `dimnames(X) <- list(dn[[1L]], unlist(collabs[nc > 0], use.names = FALSE))`:
! length of 'dimnames' [2] not equal to array extent
# A tibble: 0 × 1
# ℹ 1 variable: match_id <int>

`

For the record, using Fotmob API directly works fine so it isn't an API issue. I've reinstalled, restarted, and otherwise tried all options to resolve the error.

Using Mac OS, package version 0.6.3.3.

### PR DESCRIPTION
**Pull request by tonyelhabr**
Fixes #277. I think the issue was due to additional nesting of some elements under the `stats` element (e.g. `rating`), although I don't know for sure since I can't track the history of the API.

Anyways, I decided it was time to do a bit of refactoring of the internals of `fomtob_get_match_players()`. It's necessarily messy, and I improved it a bit, although I'd still consider it messy. (It's by far the hardest things to parse out of their API.)

### CODE REVIEW

File: R/fotmob_players.R [hunk 1]
```diff
@@ -70,147 +186,21 @@ fotmob_get_match_players <- function(match_ids) {
...
-      player_ids <- as.character(p[["id"]])
-      n <- length(player_ids)
-      ## for some reason jsonlite tries to combine stuff row-wise when it really should just bind things column-wise
-      ps <- p[["stats"]]
-
-      .clean_stats <- function(x) {
-        if(is.null(x)) {
-          return(data.frame(dummy = 1))
-        }
-        xt <- t(x)
```
Comment (tonyelhabr) [inline R/fotmob_players.R:85]: issue was here. the whole `.clean_positions()` was sort of janky, so improved it by re-converting the dataframes to lists using `jsonlite::fromJSON(jsonlite::toJSON())`

Comment (tonyelhabr) [reply R/fotmob_players.R:85]: aside from the change to the internals of `.clean_positions()`, there aren't many actual code changes. mostly just moving things around

File: NEWS.md [hunk 2]
```diff
@@ -2,13 +2,15 @@
 
 ### Bugs
 
-* `fotmob_get_match_details()` failing due to change in `teamColors` JSON element (0.6.2.3001) [#271](https://github.com/JaseZiv/worldfootballR/issues/271)
-* `fotmob_get_league_ids()` failing due to addition of `localizedName` JSON element (0.6.2.3003) [#27](https://github.com/JaseZiv/worldfootballR/issues/275)
-* `fotmob_get_match_details()` failing (again) due to change in `teamColors` JSON element (0.6.2.3003)
+* `fotmob_get_match_details()` failing due to change in `teamColors` JSON element (0.6.3.0001) [#271](https://github.com/JaseZiv/worldfootballR/issues/271)
+* `fotmob_get_league_ids()` failing due to addition of `localizedName` JSON element (0.6.3.0003) [#275](https://github.com/JaseZiv/worldfootballR/issues/275)
+* `fotmob_get_match_details()` failing (again) due to change in `teamColors` JSON element (0.6.3.0003)
```
Comment (tonyelhabr) [inline NEWS.md:7]: drive by fix version numbers from past PRs (`0.6.2.300x` -> `0.6.3.000x`)

File: tests/testthat/test-fotmob.R [hunk 3]
```diff
@@ -461,7 +461,7 @@ test_that("fotmob_get_match_players() works", {
...
..., "usual_position", "position_row", "role", "is_captain", "subbed_out", "g", "rating_num", "rating_bgcolor", "is_top_rating", "is_match_finished", "fantasy_score_num", "fantasy_score_bgcolor", "home_team_id", "home_team_color", "away_team_id", "away_team_color", "stats_fot_mob_rating", "stats_minutes_played", "stats_saves", "stats_goals_conceded", "stats_x_got_faced", "stats_accurate_passes", "stats_accurate_long_balls", "stats_diving_save", "stats_saves_inside_box", "stats_acted_as_sweeper", "stats_punches", "stats_throws", "stats_high_claim", "stats_recoveries", "stats_touches", "stats_goals", "stats_assists", "stats_total_shots", "stats_chances_created", "stats_expected_assists_x_a", "stats_successful_dribbles", "stats_accurate_crosses", "stats_dispossessed", "stats_tackles_won", "stats_clearances", "stats_headed_clearance", "stats_interceptions", "stats_dribbled_past", "stats_ground_duels_won", "stats_aerial_duels_won", "stats_was_fouled", "stats_fouls_committed", "stats_expected_goals_x_g", "stats_shot_accuracy", "stats_blocks", "stats_corners", "stats_offsides", "stats_blocked_shots", "stats_expected_goals_on_target_x_got", "stats_big_chances_missed", "shotmap", "is_starter")
```
Comment (tonyelhabr) [inline tests/testthat/test-fotmob.R:464]: `stats_shotmap` removed. (`shotmap` is already its own column, and `stats_shotmap` was redundant). `stats_big_chance_missed` -> `stats_big_chances_missed` (renaming by Fotmob)

**Comment by tonyelhabr** [comment]
@JaseZiv failing test are related to transfermarket (see #278)

**Review by JaseZiv — COMMENTED** [review]
lgtm

**Review by JaseZiv — APPROVED** [review]

**Pull request merged**